File: CS/www/Controls/HeaderSearch.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

public partial class Controls_HeaderSearch : System.Web.UI.UserControl
{
    protected void Page_Load(object sender, EventArgs e)
    {

    }

    protected void cmdSearch_Click(object sender, ImageClickEventArgs e)
    {
        Response.Redirect("/Results.aspx?Search=" + HttpUtility.UrlEncode(txtSearch.Text.Trim()), true);
    }
}

File: legacy/Deploy/www/Controls/HeaderSearch.ascx.vb

Partial Class Controls_HeaderSearch
    Inherits System.Web.UI.UserControl

    Protected Sub cmdSearch_Click(ByVal sender As Object, ByVal e As System.Web.UI.ImageClickEventArgs) 'Handles cmdSearch.Click
        Response.Redirect("/Results.aspx?Search=" + HttpUtility.UrlEncode(txtSearch.Text.Trim()), True)
    End Sub
End Class
